File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> vect;
        unordered_map<string,vector<int>> map;
        for(int i=0;i<strs.size();i++){
            string s=strs[i];
            sort(s.begin(),s.end());
            map[s].push_back(i);
        }
        for(auto &entry : map) {
            vector<string> curr;
            for(int index : entry.second) {
                curr.push_back(strs[index]);
            }
            vect.push_back(curr);
        }

        return vect;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        List<List<String>> list=new ArrayList<>();
        HashMap<String,List<Integer>> map=new HashMap<>();
        for(int i=0;i<strs.length;i++){
            String s = strs[i];
            char[] arr = s.toCharArray();
            Arrays.sort(arr);
            String sorted = new String(arr);
            List<Integer> indices = map.getOrDefault(sorted, new ArrayList<>());
            indices.add(i);
            map.put(sorted, indices);
        }
        for (List<Integer> indices : map.values()) {
            List<String> temp = new ArrayList<>();

            for (int index : indices) {
                temp.add(strs[index]);
            }
            list.add(temp);
        }
        return list;
    }
}